### ISSUE TITLE
Add a REPL shortcut for `:t` with implicits shown

### ISSUE THREAD

**Issue by youngmysteriouslight** [idris-lang/Idris2#1241]
# Steps to Reproduce

```idris
data V : Nat -> Type

fpiei : (f : {e : Nat} -> V e) -> V e
fp0ei : (0 f : {e : Nat} -> V e) -> V e
fpie0 : (f : {0 e : Nat} -> V e) -> V e
fp0e0 : (0 f : {0 e : Nat} -> V e) -> V e
fmiei : {e : Nat} -> (f : V e) -> V e
fm0ei : {e : Nat} -> (0 f : V e) -> V e
fmie0 : {0 e : Nat} -> (f : V e) -> V e
fm0e0 : {0 e : Nat} -> (0 f : V e) -> V e
```
(naming convention: `fpie0` stands for "argument `f` is polymorphic, its parametricity is infinity, parametricity of `e` is zero", `fm0ei` stands for "monomorphic `f` with zero parametricity, parametricity of `e` is infinity")

Run in REPL
```
Main> :t fpiei
Main.fpiei : V e -> V e
Main> :t fpie0
Main.fpie0 : V e -> V e
Main> :t fmiei
Main.fmiei : V e -> V e
Main> :t fmie0
Main.fmie0 : V e -> V e
Main> :t fp0ei
Main.fp0ei : (0 _ : V e) -> V e
Main> :t fp0e0
Main.fp0e0 : (0 _ : V e) -> V e
Main> :t fm0ei
Main.fm0ei : (0 _ : V e) -> V e
Main> :t fm0e0
Main.fm0e0 : (0 _ : V e) -> V e
```

# Expected Behavior

`:t` shows different types distinguishably.

# Observed Behavior

4 different types are represented with `V e -> V e`.

**Comment by youngmysteriouslight** [comment]
The example points out that `:t` ignores implicit parameters. That can confuse while debugging since `V e -> V e` is interpreted as `{e : ?} -> V e -> V e` but never `({e : ?} -> V e) -> V e`.

I think the problem might be in Resugar.idr module which begins with the following note:
https://github.com/idris-lang/Idris2/blob/ec77ad21abfcc335971c8170e3dbe94372f86431/src/Idris/Resugar.idr#L20
```
-- Convert checked terms back to source syntax. Note that this is entirely
-- for readability therefore there is NO GUARANTEE that the result will
-- type check (in fact it probably won't due to tidying up names for
-- readability).
```

**Comment by gallais** [comment]
This is by design. You can use `:set showimplicits` if you want more
details about the binding sites of implicits.

**Comment by youngmysteriouslight** [comment]
To be clear: `:t` shows a type without implicits by default, but if `:set showimplicits` option is on, it shows the exact type.

That sounds reasonable. It must be written in docs/tutorial.

**Comment by mattpolzin** [comment]
Maybe Idris could introduce a `:t+` a la `psql` that showed implicits regardless of `:set showimplicits`. 

Awkward part is that `:t` is supposed to just be shorthand for `:type` so it's a bit weird to give it a distinct variant `:t+`. 

I do realize this could deserve discussion in its own ticket; I just couldn't help put the idea down in writing when it came to me.

**Comment by gallais** [comment]
Fixed by #1338
